src/Hosting/TestHost/src/HttpContextBuilder.cs

```diff
@@ -136,6 +142,7 @@ internal void ClientInitiatedAbort()
             }
             // Writes will still succeed, the app will only get an error if they check the CT.
             _responseReaderStream.Abort(new IOException("The client aborted the request."));
+            _clientAbortCallback?.Invoke();
```

Comment:
What about server initiated aborts? Or if the app just completes without completely reading the request body?
https://github.com/aspnet/AspNetCore/blob/0da5fabdbdd462cdd9528398efb6b1ffb9b27f65/src/Hosting/TestHost/src/HttpContextBuilder.cs#L98-L105

Reply:
I've added a unit test for the server not reading request and server abort.